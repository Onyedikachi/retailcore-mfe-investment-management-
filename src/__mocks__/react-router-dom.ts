export const useParams = jest.fn();
export const useLocation = jest.fn();
