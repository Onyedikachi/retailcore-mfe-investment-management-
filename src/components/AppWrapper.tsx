import { Provider } from "react-redux";
import { PersistGate } from "redux-persist/integration/react";
import { persistStore } from "redux-persist";
import { ToastContainer, Zoom } from "react-toastify";
import { store } from "../config/store";

function AppWrapper({ children }) {
  let persistor = persistStore(store());

  return (
    <div data-testid="app">
      <Provider data-testid="store" store={store()}>
        <PersistGate loading={<div>loading...</div>} persistor={persistor}>
          <ToastContainer
            position="top-right"
            autoClose={4000}
            draggable={false}
            // icon={false}
            transition={Zoom}
          />
          {children}
        </PersistGate>
      </Provider>
    </div>
  );
}

export default AppWrapper;
