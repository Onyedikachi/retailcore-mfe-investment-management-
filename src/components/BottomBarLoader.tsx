import React from "react";

const BottomBarLoader = () => {
  return (
    <div
      data-testid="bottom-bar-loader"
      className="w-full  p-2 flex justify-center items-center"
    >
      <div className="spinner-border h-10 w-10 border-t border-danger-500 rounded-full animate-spin"></div>
    </div>
  );
};

export default BottomBarLoader;
