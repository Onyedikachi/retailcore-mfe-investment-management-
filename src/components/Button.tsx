import { forwardRef } from "react";
import clsx from "clsx";
import { BiLoaderAlt } from "react-icons/bi";

type Ref = HTMLButtonElement;

export type ButtonProps = React.DetailedHTMLProps<
  React.ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
> & {
  as?: React.ElementType;
  isLoading?: boolean;
  disabled?: boolean;
  type?: "button" | "submit" | "reset"; // Add the default value for type prop
};

const Button = forwardRef<Ref, ButtonProps>((props, ref) => {
  const {
    as: Component = "button", // Use "button" as default if as prop is not provided
    type = "button",
    className,
    children,
    isLoading,
    disabled,
    ...rest
  } = props;
  const merged = clsx("btn", className);

  return (
    <Component
      ref={ref}
      type={type}
      className={merged}
      {...rest}
      disabled={disabled}
    >
      {children}
      {isLoading && (
        <span>
          <BiLoaderAlt data-testid="loader" className="animate-spin" />
        </span>
      )}
    </Component>
  );
});

Button.displayName = "Button";
export default Button;
