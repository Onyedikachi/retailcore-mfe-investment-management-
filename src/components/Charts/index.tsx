export { default as DonutChart } from './DonutChart'
export { default as OverviewDonutChartInfo } from './OverviewDonutChartInfo'
export {default as InvestmentVolume} from './InvestmentVolume'