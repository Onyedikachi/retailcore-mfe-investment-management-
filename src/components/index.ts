export { default as Breadcrumbs } from "./Breadcrumbs";
export { default as DropDown } from "./DropDown";
export { default as Button } from "./Button";
export { default as Loader } from "./Loader";
export { default as MultiRangeSlider } from "./MultiRangeSlider";
export { default as ProcessingStatusSlider } from "./ProcessingStatusSlider";
export { default as TopBar } from "./TopBar";
export { default as StatusCard } from "./StatusCard";
export { default as TableComponent } from "./investment/dashboard/TableComponent";
export { default as QuickLinks } from "./QuickLinks";
export { default as FormStepComponent } from "./FormStepComponent";
export { default as ProductSearch } from "./ProductSearch";
