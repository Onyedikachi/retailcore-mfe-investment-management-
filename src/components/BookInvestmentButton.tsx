import React, { useState } from "react";
import OutsideClickHandler from "react-outside-click-handler";
import { BookInvestmentOptions, IBookInvestmentOptions } from "@app/constants";
import { useNavigate } from "react-router-dom";

type BookInvestmentButton = {
  children?: any;
  disabled?: boolean;
};

export default function BookInvestmentButton({
  children,
  disabled,
}: BookInvestmentButton) {
  const navigate = useNavigate();
  const [isOpen, setIsOpen] = useState(false);

  const goToUrl = (url) => {
    if (disabled) return;
    navigate(url);

    setIsOpen(false);
  };

  return (
    <div>
      <OutsideClickHandler
        onOutsideClick={() => {
          setIsOpen(false);
        }}
      >
        <div className="relative">
          <div onKeyUp={() => {}} onClick={() => setIsOpen(true)}>{children}</div>
          {isOpen && (
            <div className="top-0 absolute w-[286px] z-10 bg-white rounded-b-lg shadow-[0px_0px_4px_0px_rgba(0,0,0,0.25)] pt-[1px]">
              <ul>
                {BookInvestmentOptions?.map((item: IBookInvestmentOptions) => (
                  <li className="group" key={item.title}>
                    <button
                      disabled={disabled}
                      onClick={() => {
                        goToUrl(item.url);
                      }}
                      className={`disabled:opacity-50 disabled:cursor-not-allowed w-full z-10 relative text-base text-[#636363] capitalize pl-6 pr-4 flex justify-between items-center py-3 group-last:rounded-b-lg hover:bg-[#F9E5E5] cursor-pointer`}
                    >
                      {item.title}
                    </button>
                  </li>
                ))}
              </ul>
            </div>
          )}
        </div>
      </OutsideClickHandler>
    </div>
  );
}
