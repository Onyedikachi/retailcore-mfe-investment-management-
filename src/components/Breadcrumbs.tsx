import React from "react";
import { BsArrowLeft } from "react-icons/bs";
import { useNavigate } from "react-router-dom";

interface BreadcrumbItem {
  id: string | number;
  title: string;
  url: string;
}

interface BreadcrumbsProps {
  links: BreadcrumbItem[];
}

export default function Breadcrumbs({
  links,
}: BreadcrumbsProps): React.ReactElement {
  const navigate = useNavigate();
  return (
    <div className="flex items-center gap-x-2" data-testid="breadcrumbs">
      <button
        className="py-1 px-1 h-8 w-8 rounded flex items-center justify-center outline-none bg-transparent hover:bg-[#F9F2F2]"
        onClick={() => navigate(-1)}
        data-testid="back button"
      >
        <BsArrowLeft className="text-[#636363]" />{" "}
      </button>
      <ul className="flex gap-x-8 uppercase items-center">
        {links.map((item: BreadcrumbItem, idx: number) => (
          <li
            key={`${idx.toString()}-index`}
            onMouseOver={() => { }}
            onClick={() => navigate(item.url)}
            data-testid={item.title}
            className={`first-of-type:hover:bg-[#F9F2F2] ${item.url !== "#" ? "hover:bg-[#F9F2F2] cursor-pointer" : ""
              } px-2 py-1 rounded first-of-type:cursor-pointer relative text-[#8F8F8F] text-base font-medium last-of-type:text-[#636363] after:absolute after:content-[''] after:right-[-16px] after:top-1/2 after:translate-y-[-50%] after:w-2 after:h-2 after:border-b-2 after:border-r-2 after:border-[#636363] after:-rotate-45 last-of-type:after:content-none`}
          >
            <span>{item.title}</span>
          </li>
        ))}
      </ul>
    </div>
  );
}
