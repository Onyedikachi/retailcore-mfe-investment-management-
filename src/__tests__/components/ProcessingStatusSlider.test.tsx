// Breadcrumbs.test.jsx

import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { ProcessingStatusSlider } from "../../components";

const links = [
  { id: 1, title: "Home", url: "/" },
  { id: 2, title: "Branches", url: "/branches" },
];
jest.mock("react-router-dom", () => ({
  BrowserRouter: ({ children }) => <div>{children}</div>,
  Routes: ({ children }) => <div>{children}</div>,
  Route: ({ element }) => element,
}));
describe("ProcessingStatusSlider", () => {
  it("renders without crashing", () => {
    render(
      <ProcessingStatusSlider
        rangeLabels={["Pending submission", "Approved"]}
      />
    );
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <ProcessingStatusSlider
        rangeLabels={["Pending submission", "Approved"]}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe("ProcessingStatusSlider", () => {
  // Tests that the component renders with the correct UI elements and props
  it("should render the Processing Status Slider component with the correct UI elements and props", () => {
    const rangeLabels = ["Label 1", "Label 2"];
    const isDisabled = true;
    const firstRangeDisabled = true;
    const secondRangeDisabled = false;
    render(
      <ProcessingStatusSlider
        rangeLabels={rangeLabels}
        isDisabled={isDisabled}
        firstRangeDisabled={firstRangeDisabled}
        secondRangeDisabled={secondRangeDisabled}
      />
    );
    const processingStatusSlider = screen.getByText("Processing Status:");
    const label1 = screen.getByTestId("label1");
    const label2 = screen.getByTestId("label2");
    const firstRangeDisabledImage = screen.getByTestId("firstRangeDisabled");
    const secondRangeDisabledImage = screen.getByTestId("secondRangeDisabled");
    expect(processingStatusSlider).toBeInTheDocument();
    expect(label1).toHaveTextContent("Label 1");
    expect(label2).toHaveTextContent("Label 2");
    expect(firstRangeDisabledImage).toHaveAttribute("alt", "sliderLogo");
    expect(firstRangeDisabledImage).toHaveClass("opacity-40");
    expect(secondRangeDisabledImage).toHaveAttribute("alt", "sliderLogo");
    expect(secondRangeDisabledImage).not.toHaveClass("opacity-40");
  });

  // Tests that the component renders with default props
  it("should render the Processing Status Slider component with default props", () => {
    render(<ProcessingStatusSlider rangeLabels={[]} />);
    const processingStatusSlider = screen.getByText("Processing Status:");
    const label1 = screen.getByTestId("label1");
    const label2 = screen.getByTestId("label2");
    const firstRangeDisabledImage = screen.getByTestId("firstRangeDisabled");
    const secondRangeDisabledImage = screen.getByTestId("secondRangeDisabled");
    expect(processingStatusSlider).toBeInTheDocument();
    expect(label1).toHaveTextContent("");
    expect(label2).toHaveTextContent("");
    expect(firstRangeDisabledImage).toHaveAttribute("alt", "sliderLogo");
    expect(firstRangeDisabledImage).not.toHaveClass("opacity-40");
    expect(secondRangeDisabledImage).toHaveAttribute("alt", "sliderLogo");
    expect(secondRangeDisabledImage).not.toHaveClass("opacity-40");
  });

  // Tests that the component renders with all ranges enabled
  it("should render the Processing Status Slider component with all ranges enabled", () => {
    const rangeLabels = ["Label 1", "Label 2"];
    render(<ProcessingStatusSlider rangeLabels={rangeLabels} />);
    const processingStatusSlider = screen.getByText("Processing Status:");
    const label1 = screen.getByTestId("label1");
    const label2 = screen.getByTestId("label2");
    const firstRangeDisabledImage = screen.getByTestId("firstRangeDisabled");
    const secondRangeDisabledImage = screen.getByTestId("secondRangeDisabled");
    expect(processingStatusSlider).toBeInTheDocument();
    expect(label1).toHaveTextContent("Label 1");
    expect(label2).toHaveTextContent("Label 2");
    expect(firstRangeDisabledImage).toHaveAttribute("alt", "sliderLogo");
    expect(firstRangeDisabledImage).not.toHaveClass("opacity-40");
    expect(secondRangeDisabledImage).toHaveAttribute("alt", "sliderLogo");
    expect(secondRangeDisabledImage).not.toHaveClass("opacity-40");
  });

  // Tests that the component renders with all ranges disabled
  it("should render the Processing Status Slider component with all ranges disabled", () => {
    const rangeLabels = ["Label 1", "Label 2"];
    const isDisabled = true;
    const firstRangeDisabled = true;
    const secondRangeDisabled = true;
    render(
      <ProcessingStatusSlider
        rangeLabels={rangeLabels}
        isDisabled={isDisabled}
        firstRangeDisabled={firstRangeDisabled}
        secondRangeDisabled={secondRangeDisabled}
      />
    );
    const processingStatusSlider = screen.getByText("Processing Status:");
    const label1 = screen.getByTestId("label1");
    const label2 = screen.getByTestId("label2");
    const firstRangeDisabledImage = screen.getByTestId("firstRangeDisabled");
    const secondRangeDisabledImage = screen.getByTestId("secondRangeDisabled");
    expect(processingStatusSlider).toBeInTheDocument();
    expect(label1).toHaveTextContent("Label 1");
    expect(label2).toHaveTextContent("Label 2");
    expect(firstRangeDisabledImage).toHaveAttribute("alt", "sliderLogo");
    expect(firstRangeDisabledImage).toHaveClass("opacity-40");
    expect(secondRangeDisabledImage).toHaveAttribute("alt", "sliderLogo");
    expect(secondRangeDisabledImage).toHaveClass("opacity-40");
  });

  // Tests that the component renders with only the first range disabled
  it("should render the Processing Status Slider component with only the first range disabled", () => {
    const rangeLabels = ["Label 1", "Label 2"];
    const firstRangeDisabled = true;
    render(
      <ProcessingStatusSlider
        rangeLabels={rangeLabels}
        firstRangeDisabled={firstRangeDisabled}
      />
    );
    const processingStatusSlider = screen.getByText("Processing Status:");
    const label1 = screen.getByTestId("label1");
    const label2 = screen.getByTestId("label2");
    const firstRangeDisabledImage = screen.getByTestId("firstRangeDisabled");
    const secondRangeDisabledImage = screen.getByTestId("secondRangeDisabled");
    expect(processingStatusSlider).toBeInTheDocument();
    expect(label1).toHaveTextContent("Label 1");
    expect(label2).toHaveTextContent("Label 2");
    expect(firstRangeDisabledImage).toHaveAttribute("alt", "sliderLogo");
    expect(firstRangeDisabledImage).toHaveClass("opacity-40");
    expect(secondRangeDisabledImage).toHaveAttribute("alt", "sliderLogo");
    expect(secondRangeDisabledImage).not.toHaveClass("opacity-40");
  });

  // Tests that the component renders with only the second range disabled
  it("should render the Processing Status Slider component with only the second range disabled", () => {
    const rangeLabels = ["Label 1", "Label 2"];
    const secondRangeDisabled = true;
    render(
      <ProcessingStatusSlider
        rangeLabels={rangeLabels}
        secondRangeDisabled={secondRangeDisabled}
      />
    );
    const processingStatusSlider = screen.getByText("Processing Status:");
    const label1 = screen.getByTestId("label1");
    const label2 = screen.getByTestId("label2");
    const firstRangeDisabledImage = screen.getByTestId("firstRangeDisabled");
    const secondRangeDisabledImage = screen.getByTestId("secondRangeDisabled");
    expect(processingStatusSlider).toBeInTheDocument();
    expect(label1).toHaveTextContent("Label 1");
    expect(label2).toHaveTextContent("Label 2");
    expect(firstRangeDisabledImage).toHaveAttribute("alt", "sliderLogo");
    expect(firstRangeDisabledImage).not.toHaveClass("opacity-40");
    expect(secondRangeDisabledImage).toHaveAttribute("alt", "sliderLogo");
    expect(secondRangeDisabledImage).toHaveClass("opacity-40");
  });
});
