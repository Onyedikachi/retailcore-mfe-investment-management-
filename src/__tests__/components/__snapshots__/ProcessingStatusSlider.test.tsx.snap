// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProcessingStatusSlider matches snapshot 1`] = `
<DocumentFragment>
  <div
    class=" bg-[#ffffff]   border border-[#EEEEEE] rounded-[10px] px-[60px] pt-[40px] pb-[40px] "
  >
    <p
      class="text-[#636363] text-lg font-semibold"
    >
      Processing Status:
    </p>
    <div
      class="w-full  mt-6 "
    >
      <div
        class="px-4 pb-2 flex items-center gap-x-[1px]"
      >
        <img
          alt="sliderLogo"
          class=""
          data-testid="firstRangeDisabled"
          src=""
        />
        <div
          class="flex-1 bg-[#D9D9D9] h-[10px]"
        />
        <img
          alt="sliderLogo"
          class=""
          data-testid="secondRangeDisabled"
          src=""
        />
      </div>
      <div
        class="flex justify-between mt-2"
      >
        <div
          class="break-word text-[#636363] text-xs font-medium "
          data-testid="label1"
        >
          Pending submission
        </div>
        <div
          class="break-word text-[#636363] text-xs font-medium "
          data-testid="label2"
        >
          Approved
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
